packages/editor/src/client/blocks/richTextBlock.tsx

```diff
@@ -431,6 +486,139 @@ function RemoveLinkFormatButton() {
+  function BaseTableButton({icon, iconActive, active, ...props}, ref) {
+    icon = iconActive && active ? iconActive : icon
+    return (
+      <button
+        style={{
+          border: active ? 'blue 1px solid' : '',
+          fontSize: 16,
+
+          cursor: 'pointer',
+          borderRadius: 3,
+          backgroundColor: 'transparent',
+
+          padding: 2
+        }}
+        ref={ref}
+        {...props}>
+        <Icon icon={icon} element={icon} />
+      </button>
+    )
+  }
+)
+
+function TableButton({icon, iconActive}: TableButtonProps) {
+  const editor = useSlate()
+  const [nrows, setNrows] = useState(2)
+  const [ncols, setNcols] = useState(1)
+
+  const [isTableSpecsOpen, setIsTableSpecsOpen] = useState(false)
+
+  const triggerRef = useRef<any>(null)
+
+  const {t} = useTranslation()
+
+  const open = () => {
+    if (triggerRef.current && !isFormatActive(editor, BlockFormat.Table)) {
+      triggerRef.current!.open()
+      // TODO rather be done with forwardRef ?
+      setIsTableSpecsOpen(true)
+    }
+  }
+  const close = () => {
+    if (triggerRef.current) {
+      triggerRef.current!.close()
+      setIsTableSpecsOpen(false)
+    }
+  }
+
+  const tableSpecs = (
+    <Popover>
+      <div
+        style={{
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-between',
+          height: '130px'
+        }}>
+        <SetRowColNumbers key={'row'} label="rows" num={nrows} setNumber={setNrows} />
+        <SetRowColNumbers key={'col'} label="columns" num={ncols} setNumber={setNcols} />
+        <Button
+          disabled={isFormatActive(editor, BlockFormat.Table)}
+          onClick={() => {
+            Transforms.insertNodes(editor, emptyCellsTable(nrows, ncols))
+            close()
+          }}>
+          {t('insert table')}
+        </Button>
+      </div>
+    </Popover>
+  )
+
+  const emptyCellsTable = (nrows: number, ncols: number): SlateElement[] => [
+    {
+      type: BlockFormat.Table,
+      children: Array.from({length: nrows}).map(_ => ({
+        type: BlockFormat.TableRow,
+        children: Array.from({length: ncols}).map(_ => ({
+          type: BlockFormat.TableCell,
+          children: [
+            {
+              type: BlockFormat.Paragraph,
+              children: [{text: ''}]
+            }
+          ]
+        }))
+      }))
+    },
+    {
+      type: BlockFormat.Paragraph,
+      children: [{text: ''}]
+    }
+  ]
+
+  return (
+    <>
+      <Whisper placement="top" speaker={tableSpecs} ref={triggerRef} trigger="none">
+        <BaseTableButton
+          icon={isTableSpecsOpen ? iconActive || icon : icon}
+          disabled={isFormatActive(editor, BlockFormat.Table)}
+          active={isTableSpecsOpen}
+          onMouseDown={e => {
+            e.preventDefault()
+            isTableSpecsOpen ? close() : open()
+          }}
+        />
+      </Whisper>
+    </>
+  )
+}
+
+interface SetRowColNumbersProps {
```

Comment:
I think you should set the `interfaces` at the beginning of the file

Reply:
Can do that. I read somewhere that interfaces should be placed next to their implementation?

Reply:
You might be right, I think I have just always encountered the interfaces on the beginning of the file in our codebase. Maybe @roosnic1 can tell us how we do it at wepublish?